docs/howto/write-unit-tests-for-a-charm.md

```diff
@@ -162,6 +162,25 @@ def test_charm_runs(my_charm):
 If you use pytest, you should put the `my_charm` fixture in a top level `conftest.py`, as it will likely be shared between all your unit tests.
 ```
 
+## Reusing state
+
+Try to make each test an isolated test of a particular event handler. After checking the `State` that `ctx.run` returns, you shouldn't need to reuse the `State` as input to another event handler.
```

Comment:
I'm not sure about this paragraph. It's not inherently wrong to chain, like:

```python
st1 = ctx.run(ctx.on.config_changed(), st0)
assert s1...
st2 = ctx.run(ctx.on.start(), st1)
assert s2...
```

We suggest that typically each test is a testing a single event, but it doesn't have to.

It's also fine to have a state and re-use it to evaluate what happens with different events. For example, a State that's an expected stable state (maybe a fixture) and asserting that you end up with an output state regardless of event order.